core/src/main/scala/org/apache/spark/sql/execution/CachedPlanHelperExec.scala

```diff
@@ -86,7 +88,17 @@ case class CachedPlanHelperExec(childPlan: CodegenSupport, @transient session: S
           }
           alreadyGotBroadcastNode = true
         }
+        bchj transformAllExpressions {
+          case pl: ParamLiteral =>
+            session.getContextObject[mutable.Map[BroadcastHashJoinExec, ArrayBuffer[Any]]](
```

Comment:
If we can setting the CachedPlanHelperExec.NOCACHING_KEY in the code above (line 77), we don't need this check?